src/platforms/stm32/src/main.c

```diff
@@ -170,10 +201,31 @@ int main()
 
     AVM_LOGI(TAG, "Starting: %s...\n", startup_module_name);
     context_execute_loop(ctx, mod, "start", 0);
-    AVM_LOGI(TAG, "Return value: %lx", (long) term_to_int32(ctx->x[0]));
 
-    while (1)
-        ;
+    term ret_value = ctx->x[0];
+    char *ret_atom_string = interop_atom_to_string(ctx, ret_value);
+    if (ret_atom_string != NULL) {
```

Comment:
Probably okay here because we are at the end of `main`, but could show up as a memory leak in a diagnostic tool.  Best to free `ret_atom_string` when you can.

Reply:
free'd.